lang/LangPrimSource/PyrListPrim.cpp

```diff
@@ -564,43 +564,55 @@ int prEvent_IsRest(struct VMGlobals *g, int numArgsPushed)
 {
 	PyrSlot *dictslots = slotRawObject(g->sp)->slots;
 	PyrSlot *arraySlot = dictslots + ivxIdentDict_array;
+	static int isRestCount = 0;
 
 	if (isKindOfSlot(arraySlot, class_array)) {
 		PyrSlot key, typeSlot;
 		PyrSymbol *typeSym;
-		// test 'this[\type] == \rest' first
+		// easy tests first: 'this[\type] == \rest'
 		SetSymbol(&key, s_type);
 		identDict_lookup(slotRawObject(g->sp), &key, calcHash(&key), &typeSlot);
 		if(!slotSymbolVal(&typeSlot, &typeSym) && typeSym == s_rest) {
 			SetBool(g->sp, 1);
 			return errNone;
-		} else {
-			PyrObject *array = slotRawObject(arraySlot);
-			PyrSymbol *slotSym;
-			PyrSlot *slot;
-			int32 size = array->size;
-			int32 i;
-
-			slot = array->slots + 1;  // scan only the odd items
-
-			for (i = 1; i < size; i += 2, slot += 2) {
-				if (isKindOfSlot(slot, class_rest)
-				    || isKindOfSlot(slot, class_metarest)
+		};
+		// and, 'this[\isRest] == true'
+		SetSymbol(&key, s_isRest);
+		identDict_lookup(slotRawObject(g->sp), &key, calcHash(&key), &typeSlot);
+		if(IsTrue(&typeSlot)) {
+			if(++isRestCount == 1)
+				post("\nWARNING: Setting isRest to true in an event is deprecated. See the Rest helpfile for supported ways to specify rests.\n\n");
+			if(isRestCount == 100)
+				isRestCount = 0;
+			SetBool(g->sp, 1);
+			return errNone;
+		};
+		// failing those, scan slot values for a Rest instance or \ or \r
+		PyrObject *array = slotRawObject(arraySlot);
+		PyrSymbol *slotSym;
+		PyrSlot *slot;
+		int32 size = array->size;
+		int32 i;
+
+		slot = array->slots + 1;  // scan only the odd items
+
+		for (i = 1; i < size; i += 2, slot += 2) {
+			if (isKindOfSlot(slot, class_rest)
+			    || isKindOfSlot(slot, class_metarest)
+			) {
+				SetBool(g->sp, 1);
+				return errNone;
+			} else if(!slotSymbolVal(slot, &slotSym)) {
+				if(slotSym == s_empty
+					|| slotSym == s_r
+					|| slotSym == s_rest
 				) {
 					SetBool(g->sp, 1);
 					return errNone;
-				} else if(!slotSymbolVal(slot, &slotSym)) {
-					if(slotSym == s_empty
-						|| slotSym == s_r
-						|| slotSym == s_rest
-					) {
-						SetBool(g->sp, 1);
-						return errNone;
-					}
-				}  // why no 'else'?
-				// slotSymbolVal nonzero return = not a symbol;
-				// non-symbols don't indicate rests, so, ignore them.
-			}
+				}
+			}  // why no 'else'?
+			// slotSymbolVal nonzero return = not a symbol;
+			// non-symbols don't indicate rests, so, ignore them.
```

Comment:
I would like to see this (lines 600-615) moved out into a separate function if you have the time. That way you can localize `slotSym` and only have to write one return statement. `bool slotIsRestlike(PyrSlot* slot)` or similar. This would make future changes easier to understand (clarifies control flow, abstracts and names subtasks, reduces number of local variables)